homeassistant/components/agent_dvr/services.yaml

```diff
@@ -0,0 +1,48 @@
+start_recording:
```

Comment:
https://www.home-assistant.io/integrations/camera/

The camera integration already offers most of these services.  Please use these instead. We should only implement services that are device specific.

Reply:
start_recording, tells agent to start recording, not home assistant. Should I rename it?

Reply:
If it does something different than its fine to keep.  

camera implements the following:
```
Available services: enable_motion_detection, disable_motion_detection, play_stream, record, snapshot, turn_off and turn_on.
```

Reply:
The integration handles all of those apart from play_stream

Reply:
Since `camera` implements them you don't need to implement them here as the user can call `camera.turn_on` which will call `async_turn_on`.

Reply:
I removed the commands from yaml, now when I try to call the commands on the camera like start_recording, hass throws

Failed to call service camera/record. required key not provided @ data['filename']

.. that's not how agent works - it generates it's own filenames internally.

Reply:
also none of the other commands are now working - turn-off just disables the camera in hass, it's not calling turn-off on agent.

Reply:
It looks like the ones that can be handled by the camera services are `turn_on`, `turn_off`, `enable_motion_detection`, and `disable_motion_detection`.  This may not be an exhaustive list.

Reply:
enable_motion_detection and disable_motion_detection now don't do anything

Reply:
i'll just remove it, getting pretty frustrated with this!

Reply:
The camera needs to implement
`(async_)enable_motion_detection` and `(async_)disable_motion_detection` for the service to work.  If it doesn't implement these, then removing them is the right thing to do.

https://github.com/home-assistant/core/blob/dev/homeassistant/components/camera/__init__.py#L279

Reply:
our latest code does have the motion_detection stuff in it - i'll merge it in. We're aboput 6 weeks ahead on development on this so backporting it to this code base is rather painful.

Reply:
I wouldn't worry too much about add support for it right away as you can always add more in a followup PR once this one is merged.  The majority of our reviewers are volunteers (including myself).  It will be a little easier to do later because we have more reviewers that have free time blocks to digest small chunks than completely new integrations.